jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java

```diff
@@ -340,4 +338,9 @@ public TimerJobInstance getTimerByName(String jobName) {
 		return found;
 	}
 
+    public void evictCache(JobHandle jobHandle) {
+        String jobName =  ((EjbGlobalJobHandle) jobHandle).getUuid();
+        logger.debug("Invalidate job {} with job name {} in cache", jobName, localCache.remove(jobName));
```

Comment:
Shouldn't remove call be out of the logger invocation? If not in debug level, it won't be executed

Reply:
that will be always executed (it is not a function) as you need to pass it as parameter to the debug message. A whole different thing if it is logged or not.